tests/metrics/test_cuboid_metrics.py

```diff
@@ -8,4 +17,300 @@
         allow_module_level=True,
     )
 
-# TODO(gunnar): Add Cuboid tests!
+CAR_LABEL = "car"
+PEDESTRIAN_LABEL = "pedestrian"
+
+
+def test_cuboid_metrics_simple():
+    # single item, perfect predictions
+    annotations = [
+        CuboidAnnotation(
+            label=CAR_LABEL,
+            position=Point3D(0, 0, 0),
+            dimensions=Point3D(10, 10, 10),
+            yaw=0.0,
+            reference_id="item_A",
+        ),
+        CuboidAnnotation(
+            label=CAR_LABEL,
+            position=Point3D(1000, 1000, 1000),
+            dimensions=Point3D(10, 10, 10),
+            yaw=0.0,
+            reference_id="item_A",
+        ),
+    ]
+    predictions = [
+        CuboidPrediction(
+            label=CAR_LABEL,
+            position=Point3D(0, 0, 0),
+            dimensions=Point3D(10, 10, 10),
+            yaw=0.0,
+            reference_id="item_A",
+        ),
+        CuboidPrediction(
+            label=CAR_LABEL,
+            position=Point3D(1000, 1000, 1000),
+            dimensions=Point3D(10, 10, 10),
+            yaw=0.0,
+            reference_id="item_A",
+        ),
+    ]
+    assert CuboidIOU().eval(annotations, predictions) == ScalarResult(
+        1.0, len(annotations)
+    ), "Unexpected Cuboid IoU result"
+    assert CuboidPrecision().eval(annotations, predictions) == ScalarResult(
+        1.0, len(annotations)
+    ), "Unexpected Cuboid Precision result"
+    assert CuboidRecall().eval(annotations, predictions) == ScalarResult(
+        1.0, len(annotations)
+    ), "Unexpected Cuboid Recall result"
+
+
+def test_cuboid_metrics_numerical_check():
+    # single item, realistic predictions w/ matches and non-matches
+    annotations = [
+        CuboidAnnotation(
+            label=CAR_LABEL,
+            position=Point3D(0, 0, 0),
+            dimensions=Point3D(10, 10, 5),
+            yaw=0.0,
+            reference_id="item_A",
+        ),
+        CuboidAnnotation(
+            label=CAR_LABEL,
+            position=Point3D(1000, 1000, 1000),
+            dimensions=Point3D(10, 10, 10),
+            yaw=0.0,
+            reference_id="item_A",
+        ),
+        CuboidAnnotation(
+            label=CAR_LABEL,
+            position=Point3D(-100, -100, -100),
+            dimensions=Point3D(10, 10, 10),
+            yaw=0.0,
+            reference_id="item_A",
+        ),  # false negative
```

Comment:
This comment is very easy to overlook (but super helpful!). I'd do this in the future with an actual variable. 

I'd essentially split annotations and predictions into:
```
expected_matched_anns  = [...]
expected_unmatched_anns = [...]
annotations = expected_matched_anns + expected_unmatched_anns

matching_predictions = [...]
false_positive_predictions = [...]
predictions = matching_predictions + false_positive_predictions
```
This matches how I think most people read code, we're more observant of variable names than comments.

Reply:
makes sense! agreed that readability is important here - I appreciate the detailed suggestions please keep them coming 😁